server/graphql/v1/mutations/expenses.js

```diff
@@ -703,67 +705,94 @@ export async function payExpense(req, args) {
       );
     }
 
-    // If the amount of the expense is larger than i.e. $1000 enforce 2FA if the user has it turned on
-    const is2FARequiredForPayoutMethod = [PayoutMethodTypes.PAYPAL, PayoutMethodTypes.BANK_ACCOUNT].includes(
-      payoutMethodType,
+    // 2FA for payouts
+    const isTwoFactorAuthenticationRequiredForPayoutMethod = [
+      PayoutMethodTypes.PAYPAL,
+      PayoutMethodTypes.BANK_ACCOUNT,
+    ].includes(payoutMethodType);
+    const hostHasPayoutTwoFactorAuthenticationEnabled = get(host, 'settings.payoutsTwoFactorAuth.enabled', false);
+    const hostPayoutTwoFactorAuthenticationRollingLimit = get(
+      host,
+      'settings.payoutsTwoFactorAuth.rollingLimit',
+      1000000,
     );
-    const hostHasPayout2FAEnabled = get(host, 'settings.payoutsTwoFactorAuth.enabled', false);
-    const hostPayout2FAExpenseAmount = get(host, 'settings.payoutsTwoFactorAuth.expenseAmount', 100000);
-    if (
-      is2FARequiredForPayoutMethod &&
+    const useTwoFactorAuthentication =
+      isTwoFactorAuthenticationRequiredForPayoutMethod &&
       !args.forceManual &&
-      hostHasPayout2FAEnabled &&
-      expense.amount >= hostPayout2FAExpenseAmount
-    ) {
-      enforceTwoFactorAuthenticationOnPayouts(req, args.twoFactorAuthenticatorCode);
+      hostHasPayoutTwoFactorAuthenticationEnabled;
+    let currentRollingPayoutLimit;
+    if (useTwoFactorAuthentication) {
+      currentRollingPayoutLimit = await rollingPayoutLimitTwoFactorAuthentication(
+        req,
+        args.twoFactorAuthenticatorCode,
+        hostPayoutTwoFactorAuthenticationRollingLimit,
+        expense.amount,
+      );
     }
 
-    // Pay expense based on chosen payout method
-    if (payoutMethodType === PayoutMethodTypes.PAYPAL) {
-      const paypalEmail = payoutMethod.data.email;
-      let paypalPaymentMethod = null;
-      try {
-        paypalPaymentMethod = await host.getPaymentMethod({ service: 'paypal' });
-      } catch {
-        // ignore missing paypal payment method
-      }
-      // If the expense has been filed with the same paypal email than the host paypal
-      // then we simply mark the expense as paid
-      if (paypalPaymentMethod && paypalEmail === paypalPaymentMethod.name) {
-        feesInHostCurrency.paymentProcessorFeeInHostCurrency = 0;
-        await createTransactions(host, expense, feesInHostCurrency);
-      } else if (args.forceManual) {
+    // immediately set the cache
+    if (currentRollingPayoutLimit) {
```

Comment:
If we're there, it means the payment is 2FA authorized.

Why are we not setting the value already within the `rollingPayoutLimitTwoFactorAuthentication` method?

Reply:
Probably a leftover from when the PR had cache locking. I can move it back.

Reply:
A leftover from when there was cache locking in the PR. There was a big overhaul, maybe some of the structure doesn't make sense anymore. I'll move it.